common/src/main/java/com/microsoft/identity/common/internal/result/IBaseAuthenticationResult.java

```diff
@@ -0,0 +1,76 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+
+package com.microsoft.identity.common.internal.result;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import java.util.Date;
+
+/**
+ * Interface to wrap successful authentication result. When auth succeeds, token will be wrapped into the
+ * {@link IBaseAuthenticationResult}
+ */
+public interface IBaseAuthenticationResult {
```

Comment:
The more I look at this interface, the more it scares me....

It seems like it would be all too easy to look at this file and see:
1. That it is in `common`
2. That is is namespaced to `internal`

And then [incorrectly] assume that we can make changes to this file willy-nilly: that wouldn't be the case, however since changes made here would be reflected in public API.

I'm mulling-over the thought that this interface and `IAuthenticationResult` should be separate entirely....

Reply:
Let's take a backlog item to separate these interfaces and provide an adapter between them. Yes, it's extra paperwork, but I really think we're going to want this separation going forward.

Not going to 'flunk' on this, but we shouldn't deploy until its resolved in more future-proof way.

Reply:
See https://github.com/AzureAD/microsoft-authentication-library-for-android/issues/478

Reply:
I agree with the concerns raised w.r.t to the design here. I have taken the easier way out yesterday after incorporating to changes to get it working. I have made the changes via https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/320/commits/8b577226de00ecb30f01a110c1e7386a064dc870 to use an adapter instead.